runtime/oti/cfr.h

```diff
@@ -923,22 +921,12 @@ typedef struct J9CfrClassFile {
 #define CFR_MINOR_VERSION  3
 #define CFR_PUBLIC_PRIVATE_PROTECTED_MASK	(CFR_ACC_PUBLIC | CFR_ACC_PRIVATE | CFR_ACC_PROTECTED)
 
-#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-#define CFR_CLASS_ACCESS_MASK					(CFR_ACC_PUBLIC | CFR_ACC_FINAL | CFR_ACC_IDENTITY | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_ENUM | CFR_ACC_VALUE_TYPE | CFR_ACC_PRIMITIVE_VALUE_TYPE)
-#define J9_IS_CLASSFILE_VALUETYPE(classfile)     J9_ARE_ALL_BITS_SET((classfile)->accessFlags, CFR_ACC_VALUE_TYPE)
-#define CFR_INTERFACE_CLASS_ACCESS_MASK			(CFR_ACC_PUBLIC | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_VALUE_TYPE | CFR_ACC_IDENTITY)
-#else /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
-#define CFR_CLASS_ACCESS_MASK					(CFR_ACC_PUBLIC | CFR_ACC_FINAL | CFR_ACC_SUPER | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_ENUM)
-#define J9_IS_CLASSFILE_VALUETYPE(classfile)     FALSE
-#define CFR_INTERFACE_CLASS_ACCESS_MASK			(CFR_ACC_PUBLIC | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION)
-#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
-
-#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-#define J9_IS_CLASSFILE_PRIMITIVE_VALUETYPE(classfile)     J9_ARE_ALL_BITS_SET((classfile)->accessFlags, CFR_ACC_PRIMITIVE_VALUE_TYPE)
-#else /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
-#define J9_IS_CLASSFILE_PRIMITIVE_VALUETYPE(classfile) FALSE
-#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
+#define VALUE_TYPES_MAJOR_VERSION 67
+#define PREVIEW_MINOR_VERSION 65535
+#define J9_IS_CLASSFILE_VALUETYPE(classfile) ((classfile->majorVersion >= VALUE_TYPES_MAJOR_VERSION) && (PREVIEW_MINOR_VERSION == classfile->minorVersion) && J9_ARE_NO_BITS_SET(classfile->accessFlags, CFR_ACC_IDENTITY))
 
+#define CFR_CLASS_ACCESS_MASK					(CFR_ACC_PUBLIC | CFR_ACC_FINAL | CFR_ACC_IDENTITY | CFR_ACC_SUPER | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_ENUM)
```

Comment:
Even `CFR_ACC_IDENTITY` and `CFR_ACC_SUPER` share the same bit, not everyone reading this code is aware of that. So it is better to split this macro for VT and non-VT builds, so that they won't be confused by these 2 flags. i.e.

```
#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
//define CFR_CLASS_ACCESS_MASK with CFR_ACC_IDENTITY
#else /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
//define CFR_CLASS_ACCESS_MASK with CFR_ACC_SUPER
#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
```

Reply:
Okay. What about a comment to that effect instead?